### PR TITLE
[mars] Refactor default version identifier and default library setting

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
No behaviour change should happen here (at least that's the spirit), just breaking down code into functions.

### CODE REVIEW

**Comment by WalterBright** [comment]
Please break this up into smaller PRs rather than one giant one. I harangue everyone about this. Giant PRs:
1. are much much harder to review
2. are "take it or leave it" in their entirety, when some aspects may be good and others not
3. large diffs going over many, many pages make github in a browser very ssssllllllooooowwww, making them unpleasant and difficult
4. if the PR introduces a regression, it makes it much harder to identify where it went wrong
5. PRs should be narrowly focused, global kitchen sinks
6. just make each of your 5 commits its own PR

**Comment by WalterBright** [comment]
I've been repeating this point many times over the last few months, so I wrote this:
http://wiki.dlang.org/Starting_as_a_Contributor#Characteristics_of_a_Good_Pull_Request

File: test/fail_compilation/reserved_version.d [hunk 1]
```diff
@@ -0,0 +1,278 @@
+// REQUIRED_ARGS: -verrors=1000
```
Comment (MartinNowak) [inline test/fail_compilation/reserved_version.d]: Think you can use verrors=0.

**Comment by MartinNowak** [comment]
Can we please stop to scare away contributors w/ pointless bureaucracy. Those are easy to review refactorings of a very specific code area, the commits follow each other, just like it happens when you refractor sth. and see an opportunity for more improvements.
And after all you're living in different timezones, so while a single trivial PR already takes at least 0.5 or 1.5 (with review comment) days to merge, 5 unrelated, artificially splitted PRs will take 4.5 to 9.5 days.
It took me 30 minutes to review this on my tablet (no major problem to load the diff).

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**